src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,495 @@
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        override(ERC20PaymentClientBase_v1, RedeemingBondingCurveBase_v1)
+        returns (bool)
+    {
+        return interfaceId_ == type(IRedeemingBondingCurveBase_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+
+    /// @inheritdoc IFundingManager_v1
+    /// @return token_ The token address
+    function token() public view override returns (IERC20) {
+        return _token;
+    }
+
+    /// @notice Gets current static price for buying tokens
+    /// @return price_ The current buy price
+    function getStaticPriceForBuying()
+        public
+        view
+        override(BondingCurveBase_v1)
+        returns (uint)
+    {
+        return _oracle.getPriceForIssuance();
+    }
+
+    /// @notice Gets current static price for selling tokens
+    /// @return price_ The current sell price
+    function getStaticPriceForSelling()
+        public
+        view
+        override(RedeemingBondingCurveBase_v1, IRedeemingBondingCurveBase_v1)
+        returns (uint)
+    {
+        return _oracle.getPriceForRedemption();
+    }
+
+    /// @inheritdoc IFM_PC_ExternalPrice_Redeeming_v1
+    function getMaxBatchSize() external pure returns (uint maxBatchSize_) {
+        return MAX_BATCH_SIZE;
+    }
+
+    /// @inheritdoc IFM_PC_ExternalPrice_Redeeming_v1
+    function getOpenRedemptionAmount() external view returns (uint amount_) {
+        return _openRedemptionAmount;
+    }
+
+    /// @inheritdoc IFM_PC_ExternalPrice_Redeeming_v1
+    function getNextOrderId() external view returns (uint orderId_) {
+        return _nextOrderId;
+    }
+
+    /// @inheritdoc IFM_PC_ExternalPrice_Redeeming_v1
+    function getOrderId() external view returns (uint orderId_) {
+        return _orderId;
+    }
+
+    //--------------------------------------------------------------------------
+    // External Functions
+
+    // @audit-info Scenario: Project deposits collateral into the FM
+    /// @notice Allows depositing collateral to provide reserves for redemptions
+    /// @param amount_ The amount of collateral to deposit
+    function depositReserve(uint amount_) external onlyOrchestratorAdmin {
+        if (amount_ == 0) revert FM_ExternalPrice__InvalidAmount();
+
+        // Transfer collateral from sender to FM
+        IERC20(token()).safeTransferFrom(_msgSender(), address(this), amount_);
+
+        emit ReserveDeposited(_msgSender(), amount_);
+    }
+
+    //--------------------------------------------------------------------------
+    // Public Functions
+
+    /// @inheritdoc IFM_PC_ExternalPrice_Redeeming_v1
+    /// @param collateralAmount_ The amount of collateral to spend
+    function buy(uint collateralAmount_, uint minAmountOut_)
+        public
+        override(IFM_PC_ExternalPrice_Redeeming_v1, BondingCurveBase_v1)
+        onlyModuleRole(WHITELISTED_ROLE)
+        notBlacklisted
```

Comment:
This one can be removed as the ERC20 blacklist reverts automatically during transfers if an address is blacklisted

Reply:
removed